### PR TITLE
Base structure

### PR DESCRIPTION
**Pull request by malukenho**
Installation of basic components of the project:
- [x] Zf2 Skeleton
- [x] Doctrine Module
- [x] Zf2User

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+nbproject
+._*
+.~lock.*
+.buildpath
+.DS_Store
+.idea
```
Comment (Ocramius) [inline .gitignore]: Don't add your own local files

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "vendor/ZF2"]
```
Comment (Ocramius) [inline .gitmodules]: Kill this

File: LICENSE.txt [hunk 3]
```diff
@@ -0,0 +1,27 @@
+Copyright (c) 2005-2015, Zend Technologies USA, Inc.
```
Comment (Ocramius) [inline LICENSE.txt]: To be removed

File: composer.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (Ocramius) [inline composer.json:1]: To be updated

File: config/application.config.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
+<?php
+return array(
+    // This should be an array of module namespaces used in the application.
+    'modules' => array(
+        'Application',
```
Comment (Ocramius) [inline config/application.config.php]: Enable the modules

File: config/application.config.php [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    ),
+
+    // These are various options for the listeners attached to the ModuleManager
+    'module_listener_options' => array(
+        // This should be an array of paths in which modules reside.
+        // If a string key is provided, the listener will consider that a module
+        // namespace, the value of that key the specific path to that module's
+        // Module class.
+        'module_paths' => array(
```
Comment (Ocramius) [inline config/application.config.php]: Not needed if we use composer autoloading. Use an empty array, reduce the file to a minimum

File: config/autoload/README.md [hunk 7]
```diff
@@ -0,0 +1,8 @@
+About this directory:
```
Comment (Ocramius) [inline config/autoload/README.md]: Remove

File: config/autoload/global.php [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<?php
+/**
```
Comment (Ocramius) [inline config/autoload/global.php]: Remove

File: config/autoload/local.php.dist [hunk 9]
```diff
@@ -0,0 +1,15 @@
+<?php
+/**
```
Comment (Ocramius) [inline config/autoload/local.php.dist]: Remove

File: init_autoloader.php [hunk 10]
```diff
@@ -0,0 +1,54 @@
+<?php
+/**
```
Comment (Ocramius) [inline init_autoloader.php]: Remove this file: `vendor/autoload.php` is enough

File: module/Application/Module.php [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+namespace Application;
+
+use Zend\Mvc\ModuleRouteListener;
+use Zend\Mvc\MvcEvent;
+
+class Module
+{
+    public function onBootstrap(MvcEvent $e)
+    {
+        $eventManager        = $e->getApplication()->getEventManager();
```
Comment (Ocramius) [inline module/Application/Module.php]: Remove `onBootstrap`

File: module/Application/Module.php [hunk 12]
```diff
@@ -0,0 +1,39 @@
+<?php
+/**
```
Comment (Ocramius) [inline module/Application/Module.php]: Fix these docblocks - see other projects in the roave org

File: module/Application/language/ar_JO.po [hunk 13]
```diff
@@ -0,0 +1,153 @@
+msgid ""
```
Comment (Ocramius) [inline module/Application/language/ar_JO.po]: Remove these i18n files

File: public/index.php [hunk 14]
```diff
@@ -0,0 +1,17 @@
+<?php
+/**
+ * This makes our life easier when dealing with paths. Everything is relative
+ * to the application root now.
+ */
+chdir(dirname(__DIR__));
+
+// Decline static file requests back to the PHP built-in webserver
+if (php_sapi_name() === 'cli-server' && is_file(__DIR__ . parse_url($_SERVER['REQUEST_URI'], PHP_URL_PATH))) {
```
Comment (Ocramius) [inline public/index.php]: Remove this line

File: public/index.php [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+    return false;
+}
+
+// Setup autoloading
+require 'init_autoloader.php';
```
Comment (Ocramius) [inline public/index.php]: `vendor/autoload.php`

File: vendor/README.md [hunk 16]
```diff
@@ -0,0 +1 @@
+This vendor/ directory is where third-party modules should be installed.
```
Comment (Ocramius) [inline vendor/README.md]: `vendor` should be in `.gitignore` (all of `vendor`)

File: module/Application/Module.php [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+namespace Application;
+
+class Module
```
Comment (danizord) [inline module/Application/Module.php]: Implements `Zend\ModuleManager\Feature\ConfigProviderInterface`

File: module/Application/Module.php [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+{
+    public function getConfig()
+    {
+        return include __DIR__ . '/config/module.config.php';
+    }
+
+    public function getAutoloaderConfig()
```
Comment (danizord) [inline module/Application/Module.php]: I think you can remove this method and rely on composer for autoloading :)

File: module/Application/config/module.config.php [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+    'service_manager' => [
+        'abstract_factories' => [
+            'Zend\Cache\Service\StorageCacheAbstractServiceFactory',
+            'Zend\Log\LoggerAbstractServiceFactory',
+        ]
+    ],
+
+    'controllers' => [
+        'invokables' => [
+            'Application\Controller\Index' => 'Application\Controller\IndexController'
```
Comment (danizord) [inline module/Application/config/module.config.php]: I'd suggest to use FQCN as identifier

File: module/Application/view/layout/layout.phtml [hunk 20]
```diff
@@ -0,0 +1,55 @@
+<?php echo $this->doctype(); ?>
+
+<html lang="en">
+    <head>
+        <meta charset="utf-8">
+        <?php echo $this->headTitle('ZF2 '. $this->translate('Skeleton Application'))->setSeparator(' - ')->setAutoEscape(false) ?>
```
Comment (danizord) [inline module/Application/view/layout/layout.phtml]: Roave Tickets?

File: config/application.config.php [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+return [
+    'modules' => [
+        'DoctrineModule',
+        'ZfcBase',
+        'ZfcUser',
+        'Application',
+    ],
+
+    'module_listener_options' => [
+        'module_paths' => [
```
Comment (danizord) [inline config/application.config.php]: This key is useless when using composer for autoloading

File: module/Application/config/module.config.php [hunk 22]
```diff
@@ -0,0 +1,76 @@
...
+        ],
+    ],
+
+    'view_manager' => [
+        'display_not_found_reason' => true,
+        'display_exceptions'       => true,
+        'doctype'                  => 'HTML5',
+        'not_found_template'       => 'error/404',
+        'exception_template'       => 'error/index',
+        'template_map' => [
```
Comment (danizord) [inline module/Application/config/module.config.php]: Not sure, but I think this `template_map` is needless when you have your view folder in `template_path_stack`. You really don't want to map every template you have in application

**Comment by awartoft** [comment]
bower to handle JS/CSS dependencies ? http://bowerphp.org/ or just regular npm bower

**Comment by malukenho** [comment]
@macnibblet done.

File: module/Application/config/module.config.php (hunk 19 above)
Comment (malukenho) [inline module/Application/config/module.config.php]: @danizord I don't have good information to give good names for classes, yet. It's just a structure.

**Comment by awartoft** [comment]
I don't see any bower.json file in 4c84dbbe4d28445e41bf416ecd6235ff8885415c ?

**Comment by malukenho** [comment]
@macnibblet I only put it as a dependency for now.

File: module/Application/config/module.config.php (hunk 19 above)
Comment (danizord) [inline module/Application/config/module.config.php]: @malukenho just name it `Application\Controller\IndexController` to match the name of the class.

File: .gitignore [hunk 23]
```diff
@@ -0,0 +1,5 @@
+._*
+.~lock.*
+.buildpath
```
Comment (Ocramius) [inline .gitignore:3]: Consider removing these in a subsequent PR: no need for environment-specific ignores

File: .gitignore [hunk 24]
```diff
@@ -0,0 +1,5 @@
...
+.DS_Store
```
Comment (Ocramius) [inline .gitignore:4]: Same as above. This should be in your global `.gitignore`

File: composer.json [hunk 25]
```diff
@@ -0,0 +1,23 @@
...
+        "psr-4": {
+            "Application\\": "module/Application/src/Application"
+        }
+    },
+    "homepage": "http://roave.com/",
+    "require": {
+        "php": ">=5.4",
+        "beelab/bowerphp":                  "0.2.*@alpha",
+        "zf-commons/zfc-user":              "1.*",
+        "doctrine/doctrine-module":         "0.*",
```
Comment (Ocramius) [inline composer.json:19]: `0.8.*` please

File: composer.json [hunk 26]
```diff
@@ -0,0 +1,23 @@
...
+        "tickets"
+    ],
+    "autoload": {
+        "psr-4": {
+            "Application\\": "module/Application/src/Application"
+        }
+    },
+    "homepage": "http://roave.com/",
+    "require": {
+        "php": ">=5.4",
```
Comment (Ocramius) [inline composer.json:16]: We can rely on `~5.6`

File: composer.json [hunk 27]
```diff
@@ -0,0 +1,23 @@
...
+        "beelab/bowerphp":                  "0.2.*@alpha",
+        "zf-commons/zfc-user":              "1.*",
+        "doctrine/doctrine-module":         "0.*",
+        "zendframework/zendframework":      "2.3.*",
+        "zf-commons/zfc-user-doctrine-orm": "dev-master"
```
Comment (Ocramius) [inline composer.json:21]: Is there no stable version of this?

File: composer.json (hunk 26 above)
Comment (danizord) [inline composer.json:16]: Yay! So you can put ::class everywhere :D

Comment (malukenho) [inline composer.json:16]: \o/

File: module/Application/config/module.config.php [hunk 28]
```diff
@@ -0,0 +1,70 @@
...
+                        'action'        => 'index',
+                    ],
+                ],
+            ],
+        ],
+    ],
+
+    'service_manager' => [
+        'abstract_factories' => [
+            'Zend\Cache\Service\StorageCacheAbstractServiceFactory',
```
Comment (Ocramius) [inline module/Application/config/module.config.php:49]: All `::class` whenever there's a class: exposes coupling

**Comment by Ocramius** [comment]
As discussed, move feedback related changes to a new PR

**Pull request merged**